TypeScript and style tweaks on spa

TheMikeSanto opened the pull request:
* Enabled TypeScript on the Vue app and resolved existing compilation issues (#24)
* Added a handful of style rules to spa eslint config and re-indented source files (#36)
* Added `.env.example` files in `/spa` and `/api` with example environment variables (#37)

@dburleson Please take a careful look at the new setup/style. Everything builds and works correctly in my environment, but this is a pretty huge change so pulling/building/testing the branch before merge is a good idea. Also, take a look at the changes I made to the existing code to make it TypeScript friendly. Using tsc on the api app would need some different configurations, so I'll set that up another time. Let me know what you think!

dburleson commented:
@TheMikeSanto It looks great for the most part. The typescript so far appears fine. I did have some trouble getting the environment up and running due to the .env.example file having some incorrect defaults:

api/.env: 
```
CHAT_WEBHOOK_URL="https://webhook.example"
DB_HOST="db"
DB_PORT=3306
DB_USER="root"
DB_PASS="pw"
DB_DATABASE="cybertown"
JWT_SECRET="devsecret"
PORT=3000
```

spa/.env
```
CHAT_WEBHOOK_URL="https://webhook.example"
DB_HOST="db"
DB_PORT=3306
DB_USER="root"
DB_PASS="pw"
DB_DATABASE="cybertown"
JWT_SECRET="devsecret"
PORT=8000

```

TheMikeSanto commented:
@dburleson I updated `/spa/.env.example` to use port 8000. I also removed the DB_* variables from there since that info is irrelevant to the front end code.
